core/src/main/scala/com/carto/analyticstoolbox/spark/spatial/OptimizeSpatial.scala

```diff
@@ -42,23 +43,27 @@ object OptimizeSpatial extends Serializable {
     // overwrite the output table
     try ssc.sql(s"DROP TABLE $outputTable;") catch { case _: AnalysisException => // e.printStackTrace() }*/
 
+    // Decide, based on column type of geometry, which parsing function to use
+    val table_df = ssc.table(sourceTable)
+    val parseGeom =
+      if (table_df.schema(geomColumn).dataType == "binary") "ST_geomFromTWKB"
```

Comment:
If it's binary, it's already a geom, see https://github.com/CartoDB/analytics-toolbox-databricks/blob/main/core/src/main/scala/com/carto/analyticstoolbox/core/package.scala#L32-L38

So `Geometry` is represented by the binary type already.